Java/androidTest/java/com/couchbase/litecore/C4DatabaseTest.java

```diff
@@ -382,11 +382,59 @@ public void testDatabaseExpired() throws LiteCoreException {
 
         assertEquals(expire, db.getExpiration(docID));
         assertEquals(expire, db.getExpiration(docID2));
-        assertEquals(expire, db.nextDocExpiration());
+        assertEquals((long)expire, db.nextDocExpiration());
 
         // TODO: DB00x - Java does not hava the implementation of c4db_enumerateExpired and c4exp_next yet.
     }
 
+    @Test
+    public void testPurgeExpiredDocs() throws LiteCoreException {
+        String docID = "expire_me";
+        createRev(docID, kRevID, kFleeceBody);
+
+        // unix time
+        Long expire = System.currentTimeMillis() / 1000 + 1;
+        db.setExpiration(docID, expire);
+
+        expire = System.currentTimeMillis() / 1000 + 2;
+        db.setExpiration(docID, expire);
+        db.setExpiration(docID, expire);
+
+        String docID2 = "expire_me_too";
+        createRev(docID2, kRevID, kFleeceBody);
+        db.setExpiration(docID2, expire);
+
+        try {
+            Thread.sleep(3 * 1000); // sleep 3 sec
+        } catch (InterruptedException e) {
+        }
+
+        int cnt = db.purgeExpiredDocs();
+
+        assertEquals(cnt, 2);
+    }
+
+    @Test
+    public void testPurgeDoc() throws LiteCoreException {
+        String docID = "purge_me";
+        createRev(docID, kRevID, kFleeceBody);
+        db.beginTransaction();
+        try {
+            db.purgeDoc(docID);
+        } catch(Exception e) {}
+        finally {
+            db.endTransaction(true);
+        }
+
+        try {
+            db.get(docID, true);
```

Comment:
To make sure that the exception is cached, probably add an assert fail after calling db.get().

Reply:
Please add `fail("NotFound Exception should be thrown")` after db.get(docID, true).